projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -3936,6 +3991,7 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         });
 
         this.columnList.reset(columns);
+        this.shouldGenerate = false;
```

Comment:
Shouldn't we only do this if data.length > 0 ?

This is called on content init when there may not be data yet. You will make the flag false forcing users to have to set it to true again when just having a single bind from a remote source. In this case simply putting autoGenerate to true should suffice. 

The flag should only be raised again when **re**-generating the columns.

Reply:
Ok, I agree. Will add it.